src/authentication/githubServer.ts

```diff
@@ -237,27 +119,24 @@ export class GitHubServer {
 		this.hostUri = vscode.Uri.parse(host);
 	}
 
-	public async login(): Promise<IHostConfiguration> {
-		return new Client(this.hostConfiguration.host, SCOPES)
-			.start()
-			.then(token => {
-				this.hostConfiguration.token = token;
-				return this.hostConfiguration;
-			});
-	}
-
-	public async checkAnonymousAccess(): Promise<boolean> {
-		const options = GitHubManager.getOptions(this.hostUri, 'GET', '/rate_limit');
-		return new Promise<boolean>((resolve, _) => {
-			const get = https.request(options, res => {
-				resolve(res.statusCode === 200);
-			});
-
-			get.end();
-			get.on('error', err => {
-				resolve(false);
-			});
-		});
+	public login(): Promise<IHostConfiguration> {
+		const host = this.hostUri.toString()
+		const uri = vscode.Uri.parse(
+			`${AUTH_RELAY_SERVER}/authorize?authServer=${host}&callbackUri=${CALLBACK_URI}&scope=${SCOPES}`
+		);
+		vscode.commands.executeCommand('vscode.open', uri);
+		return new Promise<IHostConfiguration>((resolve, reject) => {
+			const subscription = uriHandler.event(async uri => {
+				if (uri.path !== CALLBACK_PATH) return
+				const rsp = await axios.get(`${AUTH_RELAY_SERVER}/verify?${uri.query}`)
```

Comment:
`axios.get` can throw exceptions, in which case we want to reject the promise as well.

Reply:
Done (in the adapter logic).